airflow/models/taskinstance.py

```diff
@@ -845,8 +845,9 @@ def _refresh_from_db(
 
     :meta private:
     """
-    if session and task_instance in session:
-        session.refresh(task_instance, TaskInstance.__mapper__.column_attrs.keys())
+    if not InternalApiConfig.get_use_internal_api():
+        if session and task_instance in session:
```

Comment:
```suggestion
        if task_instance in session:
```

if we're checking use internal api, then we can probably skip checking checking session not None

because that's how it was before AIP-44

probably that condition was added before we added TracebackSession which probably evalueates to True

Reply:
Makes sense

Reply:
Actually - this also helps mypy to do their job, so let's leave it.